libs/filer/workspace_fuse_client.go

```diff
@@ -0,0 +1,218 @@
+package filer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path"
+	"slices"
+	"strings"
+
+	"github.com/databricks/cli/libs/notebook"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/workspace"
+	"golang.org/x/exp/maps"
+)
+
+type workspaceFuseClient struct {
+	workspaceFilesClient Filer
+}
+
+func stripNotebookExtension(ctx context.Context, w Filer, name string) (stripName string, stat fs.FileInfo, ok bool) {
+	ext := path.Ext(name)
+
+	extensionsToLanguages := map[string]workspace.Language{
+		".py":    workspace.LanguagePython,
+		".r":     workspace.LanguageR,
+		".scala": workspace.LanguageScala,
+		".sql":   workspace.LanguageSql,
+		".ipynb": workspace.LanguagePython,
+	}
+
+	stripName = strings.TrimSuffix(name, ext)
+
+	// File name does not have an extension associated with Databricks notebooks, return early.
+	if !slices.Contains(maps.Keys(extensionsToLanguages), ext) {
+		return "", nil, false
+	}
+
+	// If the file could be a notebook, check it's a notebook and has the correct language.
+	stat, err := w.Stat(ctx, stripName)
+	if err != nil {
+		return "", nil, false
+	}
+	info := stat.Sys().(workspace.ObjectInfo)
+
+	// Not a notebook. Return early.
+	if info.ObjectType != workspace.ObjectTypeNotebook {
+		return "", nil, false
+	}
+
+	// Not the correct language. Return early.
+	if info.Language != extensionsToLanguages[ext] {
+		return "", nil, false
```

Comment:
Both this and the return above are worth a debug log message.

Reply:
Done